src/components/molecules/AdminManagementForm.tsx

```diff
@@ -0,0 +1,188 @@
+import { observer } from 'mobx-react-lite';
+import React from 'react';
+import DialogWithHeader from './DialogWithHeader';
+import { useTranslation } from 'react-i18next';
+import { makeStyles, withStyles, createStyles } from '@material-ui/core/styles';
+import { Typography } from 'components/atoms';
+import { useStore } from 'store/storeConfig';
+import RootStore from 'store/root.store';
+import OverlayLoader from 'components/molecules/OverlayLoader';
+import { silverGrayColor } from 'style';
+import {
+  TableContainer,
+  Table,
+  Box,
+  InputLabel,
+  MenuItem,
+  Select,
+  FormControl,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableRow,
+  Theme,
+  ButtonGroup,
+  Button,
+} from '@material-ui/core';
+import { blackColor } from 'style';
+import {
+  cancelEditModeHelper,
+  changeEditObjectHelper,
+  changeCurrentSelectedRoleHelper,
+  IeditObjList,
+  IProps,
+  labels,
+} from './AdminManagementFormHelpers';
+const useStyles = makeStyles((theme: Theme) => ({
+  table: {
+    border: `1px solid ${blackColor}`,
+    borderBottom: 0,
+    borderCollapse: 'separate',
+  },
+  saveButton: {
+    marginTop: '20px',
+    marginBottom: '10px',
+  },
+}));
+const StyledTableCell = withStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      textAlign: 'center',
+      padding: theme.spacing(0.7, 0),
+      borderBottom: `1px solid ${blackColor}`,
+    },
+    sizeSmall: {
+      '&:last-child': {
+        padding: theme.spacing(0.7, 1),
+      },
+    },
+    head: {
+      backgroundColor: silverGrayColor,
+      color: theme.palette.common.black,
+    },
+  }),
+)(TableCell);
+const StyledTableRow = withStyles(() =>
+  createStyles({
+    root: {
+      // styles can be added here
+    },
+  }),
+)(TableRow);
+const AdminManagementForm: React.FC<IProps> = ({ saveEditModeHelper, isShowing, onClose, loading }) => {
+  const store: RootStore = useStore();
+  const [editObject, setEditObject] = React.useState<IeditObjList>({});
+
+  const { t } = useTranslation();
+  const classes = useStyles();
+  const EditMode: any = (props: any) => {
+    return (
+      <ButtonGroup>
+        <Button variant="contained" onClick={() => saveEditMode(props.itemData)}>
+          {t('usersManagement.save')}
+        </Button>
+        <Button variant="contained" onClick={() => cancelEditMode(props.itemData)}>
+          {t('usersManagement.cancel')}
+        </Button>
+      </ButtonGroup>
+    );
+  };
+  const ViewMode = (props: any) => {
+    return (
+      <Button onClick={() => changeEditObject(props.itemData)} variant="contained">
+        {t('usersManagement.edit')}
+      </Button>
+    );
+  };
+  const OrganizationEditList: any = ({ itemOrg, itemEmail }: any) => {
```

Comment:
please extract  `OrganizationEditList`  , `EditMode` and `ViewMode`  from the Main component (`AdminManagementForm`). you can pass props to  those components from the main component

Reply:
@danielsh28 ok, do you think it's better to open seperate file for each component/function, or is it ok to put all of them in the existing "helper" file?